dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/EndpointTests/SignIn/Register/PhoneConfirmationTests.cs

```diff
@@ -298,4 +299,112 @@ public async Task Post_BlockedClient_ReturnsTooManyRequestsStatusCode()
         // Assert
         Assert.Equal(StatusCodes.Status429TooManyRequests, (int)response.StatusCode);
     }
+
+    [Fact]
+    public async Task Post_ValidPinForNonAdminScopeWithNonAdminUser_UpdatesAuthenticationStateSignsInAndRedirects()
+    {
+        // Arrange
+        var user = await TestData.CreateUser(userType: Models.UserType.Default);
+
+        var userVerificationService = HostFixture.Services.GetRequiredService<IUserVerificationService>();
+        var pinResult = await userVerificationService.GenerateSmsPin(user.MobileNumber!);
+
+        var authStateHelper = await CreateAuthenticationStateHelper(c => c.MobileNumberSet(user.MobileNumber), additionalScopes: CustomScopes.DefaultUserTypesScopes.First());
+        var request = new HttpRequestMessage(HttpMethod.Post, $"/sign-in/register/phone-confirmation?{authStateHelper.ToQueryParam()}")
+        {
+            Content = new FormUrlEncodedContentBuilder()
+            {
+                { "Code", pinResult.Pin! }
+            }
+        };
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status302Found, (int)response.StatusCode);
+        Assert.StartsWith("/sign-in/register/phone-exists", response.Headers.Location?.OriginalString);
+
+        Assert.True(authStateHelper.AuthenticationState.MobileNumberVerified);
+        Assert.NotNull(authStateHelper.AuthenticationState.UserId);
+    }
+
+    [Fact]
+    public async Task Post_ValidPinForAdminScopeWithAdminUser_UpdatesAuthenticationStateSignsInAndRedirects()
```

Comment:
Should we even allow requests with admin scopes to enter register journey?

Reply:
We'll need a set of filters for `/Trn` and `/Register` eventually that ensure the initial request is 'compatible'. We're unlikely to support registering admin users so I don't think we need this scenario.